### ISSUE TITLE
Resume/Block uthreads example

### ISSUE THREAD

**Issue by elliott-beach** [elliott-beach/abc#1]
z0rk:
> We're missing a resume and block uthreads example and we're pretty much all set

**Comment by elliott-beach** [comment]
So this means we need to write code to for threads to block using asynchronous IO?

**Comment by carlosferry** [comment]
Correct. We need the blocking/resuming. I can write these if you wish .. I'm also re-organizing the code based on the format given.
